packages/api/test/src/routes/leaderboard.spec.ts

```diff
@@ -0,0 +1,259 @@
+          expect(ranches.length).toBe(6)
+
+          ranches.forEach((ranch, index) => {
+            expect(ranch.name).toBeTruthy()
+            expect(ranch.score).toBe(0)
+            expect(ranch.position).toBe(index)
+          })
+        }
+      )
+    })
+
+    it('players', async () => {
+      const token = await authenticatePlayer(initialPlayers[0].key)
+
+      await serverInject(
+        {
+          method: 'GET',
+          url: '/leaderboard',
+          headers: {
+            Authorization: token,
+          },
+        },
+        (err, response) => {
+          const players = response.json().players
+          expect(err).toBeFalsy()
+          expect(response.statusCode).toBe(200)
+          expect(players.length).toBe(10)
+
+          players.forEach((player, index) => {
+            expect(player.username).toBeTruthy()
+            expect(player.points).toBe(0)
+            expect(player.position).toBe(index)
+          })
+        }
+      )
+    })
+  })
+
+  describe('should return leaderboard values for each entity AFTER TRADE', () => {
+    it('bufficorns', async () => {
+      const token0 = await authenticatePlayer(initialPlayers[0].key)
+      const token1 = await authenticatePlayer(initialPlayers[1].key)
+
+      // Trade with player 1
+      await serverInject(
+        {
+          method: 'POST',
+          url: '/trades',
+          payload: {
+            to: initialPlayers[1].key,
+          },
+          headers: {
+            Authorization: token0,
+          },
+        },
+        (err, response) => {
+          expect(response.statusCode).toBe(200)
+        }
+      )
+
+      // wait until trade is over to calculate points
+      // await sleep(1300)
+
+      let selectedBufficorn
+      // Get player information to know their selected bufficorn
+      await serverInject(
+        {
+          method: 'GET',
+          url: `/players/${initialPlayers[1].key}`,
+          headers: {
+            Authorization: token1,
+          },
+        },
+        (err, response) => {
+          expect(err).toBeFalsy()
+          expect(response.statusCode).toBe(200)
+          expect(response.headers['content-type']).toBe(
+            'application/json; charset=utf-8'
+          )
+
+          selectedBufficorn = response.json().player.selectedBufficorn
+        }
+      )
+
+      await serverInject(
+        {
+          method: 'GET',
+          url: '/leaderboard',
+          headers: {
+            Authorization: token1,
+          },
+        },
+        (err, response) => {
+          const bufficorns = response.json().bufficorns
+
+          expect(err).toBeFalsy()
+          expect(response.statusCode).toBe(200)
+          expect(bufficorns.length).toBe(24)
+
+          // TODO: this test will break when we update the bufficorn names to not include a number
+          const fedBufficorn = bufficorns.find(
+            (b) => Number(b.name.split('-')[1]) === selectedBufficorn
+          )
```

Comment:
why dont use the bufficorn index instead of the split of the name?

Reply:
Because AFAIK we don't have the bufficorn index in the leaderboard response. I'm going to double check it

Reply:
should we include in case of not to be there?